src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -264,7 +264,11 @@ private function safelyUnserialize(string $serializedToken)
         });
 
         try {
-            $token = unserialize($serializedToken);
+            try {
+                $token = unserialize($serializedToken);
+            } catch (\UnserializationFailedException $e) {
+                throw 0x37313BC !== $e->getPrevious()->getCode() ? $e : $e->getPrevious();
```

Comment:
In this method, we register an "unserialize_callback_func" to gracefully detect class-not-found. The RFC breaks the logic and we need those extra lines to restore the expected behavior.